### PR TITLE
kcptun: update to version 20230207

### PR DESCRIPTION
**Pull request by liudf0716**
Maintainer: Dengfeng Liu <liudf0716@gmail.com>, Chao Liu <expiron18@gmail.com>
Compile tested: x86_64, OpenWrt 22.03-SNAPSHOT r19160+900-7ea2f3d6e2
Run tested: x86_64, OpenWrt 22.03-SNAPSHOT r19160+900-7ea2f3d6e2

Description:

add support for port-range dailer, port-range listener

Signed-off-by: Dengfeng Liu <liudf0716@gmail.com>

### CODE REVIEW

File: net/kcptun/Makefile [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-PKG_VERSION:=20210922
-PKG_RELEASE:=$(AUTORELEASE)
+PKG_VERSION:=20230207
+PKG_RELEASE:=2
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=https://codeload.github.com/xtaci/kcptun/tar.gz/v${PKG_VERSION}?
-PKG_SOURCE_DATE:=2021-09-22
-PKG_HASH:=f6a08f0fe75fa85d15f9c0c28182c69a5ad909229b4c230a8cbe38f91ba2d038
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
```
Comment (1715173329) [inline net/kcptun/Makefile:7]: uh? why?

Comment (liudf0716) [reply net/kcptun/Makefile:7]: @1715173329 
because the latest release version cannot be found in this PKG_SOURCE_URL

Comment (1715173329) [reply net/kcptun/Makefile:7]:
```console
SHELL= flock /home/immortalwrt/openwrt-master/tmp/.kcptun-20230207.tar.gz.flock -c '  	/home/immortalwrt/openwrt-master/scripts/download.pl "/home/immortalwrt/openwrt-master/dl" "kcptun-20230207.tar.gz" "skip" "" "https://codeload.github.com/xtaci/kcptun/tar.gz/v20230207?"    '
+ curl -f --connect-timeout 20 --retry 5 --location https://codeload.github.com/xtaci/kcptun/tar.gz/v20230207?/kcptun-20230207.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2545k    0 2545k    0     0  2502k      0 --:--:--  0:00:01 --:--:-- 2502k
touch /home/immortalwrt/openwrt-master/build_dir/target-aarch64_generic_musl/kcptun-20230207/.prepared_4a6dd31649740a5e7fd8bb3f5dc0a447_6664517399ebbbc92a37c5bb081b5c53_check
```

seems to work for me

===

don't name PKG_SOURCE as `v$(PKG_VERSION).tar.gz` only, it's really unrecognizable and can cause (potential) conflict.
btw unused `PKG_SOURCE_DATE` can be removed.

File: net/kcptun/Makefile [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
-PKG_RELEASE:=$(AUTORELEASE)
+PKG_VERSION:=20230207
+PKG_RELEASE:=2
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=https://codeload.github.com/xtaci/kcptun/tar.gz/v${PKG_VERSION}?
-PKG_SOURCE_DATE:=2021-09-22
-PKG_HASH:=f6a08f0fe75fa85d15f9c0c28182c69a5ad909229b4c230a8cbe38f91ba2d038
+PKG_SOURCE:=v20230207.tar.gz
+PKG_SOURCE_URL:=https://github.com/xtaci/kcptun/archive/refs/tags/
```
Comment (1715173329) [inline net/kcptun/Makefile:8]:
```suggestion
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://github.com/xtaci/kcptun/archive/refs/tags/v$(PKG_VERSION).tar.gz?
```

So, this is what I actually meant.
See http://sources.openwrt.org/, only God knows what those version.tar.gz files are.

Comment (liudf0716) [reply net/kcptun/Makefile:8]: @1715173329  I agree with u. 
but the name comes from the repo of [PKG_SOURC_URL](https://github.com/xtaci/kcptun/archive/refs/tags/), I can't find the latest version in the repo of http://sources.openwrt.org/

Comment (1715173329) [reply net/kcptun/Makefile:8]: sources is the mirror, all source files downloaded by the openwrt buildbot will be uploaded there.
Of course, this commit is not applied yet, so new tarball doesn't exist on it.

See my suggestion, use `?`in URL  to override the original file name.

And codeload still works for me, could you check your network issue etc.?

File: net/kcptun/Makefile [hunk 3]
```diff
@@ -1,13 +1,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=kcptun
-PKG_VERSION:=20210922
-PKG_RELEASE:=$(AUTORELEASE)
+PKG_VERSION:=20230207
+PKG_RELEASE:=2
```
[the remark this answers is not in the document]
Comment (liudf0716) [reply net/kcptun/Makefile]: @1715173329 ok

File: net/kcptun/Makefile (hunk 2 above)
Comment (liudf0716) [reply net/kcptun/Makefile]: @1715173329
I have changed the [PKG_SOURC_URL](https://github.com/xtaci/kcptun/archive/refs/tags/) to origin URL. the reason why I changed was that I can't download the source in my SKD environment when I submit the PR. but now it's ok.

Comment (1715173329) [reply net/kcptun/Makefile]: okay, thanks for the clarification.

File: net/kcptun/files/kcptun.init [hunk 4]
```diff
@@ -150,7 +150,7 @@ validate_common_options() {
 
 validate_server_options() {
 	validate_common_options server "$@" \
-		'listen:port' \
+		'listen:string:29900-30000' \
```
Comment (1715173329) [inline net/kcptun/files/kcptun.init:153]:
```suggestion
		'listen:or(port,portrange):29900-30000' \
```

maybe a better choice ;)

Comment (liudf0716) [reply net/kcptun/files/kcptun.init:153]: @1715173329 excellent!

**Review by 1715173329 — APPROVED** [review]

**Comment by expiron** [comment]
@liudf0716, the description about 'listen' option in the README.md needs to be updated at the same time.

**Pull request merged**